core/DeviceDetector/DeviceDetectorFactory.php

```diff
@@ -18,23 +18,33 @@ class DeviceDetectorFactory
     /**
      * Returns a Singleton instance of DeviceDetector for the given user agent
      * @param string $userAgent
+     * @param bool $useFileCache
      * @return DeviceDetector
      */
-    public static function getInstance($userAgent)
+    public static function getInstance($userAgent, $useFileCache = true)
     {
         $userAgent = Common::mb_substr($userAgent, 0, 500);
 
         if (array_key_exists($userAgent, self::$deviceDetectorInstances)) {
             return self::$deviceDetectorInstances[$userAgent];
         }
 
-        $deviceDetector = new DeviceDetector($userAgent);
-        $deviceDetector->discardBotInformation();
-        $deviceDetector->setCache(new DeviceDetectorCache(86400));
-        $deviceDetector->parse();
+        if ($useFileCache && DeviceDetectorCacheEntry::isCached($userAgent)) {
+            $deviceDetector = new DeviceDetectorCacheEntry($userAgent);
```

Comment:
not quite understanding what `discardBotInformation()` does but do we need to call it as well?